Random testing using config=Config(swarmSwitch=None, maxMustHitBranch=None, verbose=False, reducePool=False, failedLogging=None, exploitBestHint=1, greedyStutter=False, internal=False, exploit=None, seed=None, generalize=False, fastQuickAnalysis=False, stopWhenNoCoverage=None, useHints=False, localize=False, total=False, speed='FAST', probs=None, normalize=False, verboseExploit=False, highLowSwarm=None, replayable=False, essentials=False, maxMustHitStatement=None, noCoverageExploit=False, quickTests=False, relax=False, startExploitStall=0, enumerateEnabled=False, uniqueValuesAnalysis=False, swarm=False, noReassign=False, html=None, stopWhenHitStatement=None, biasLOC=None, progress=False, postCover=False, swarmLength=None, generateLOC=None, noSwarmDependencies=False, profile=False, LOCBaseline=0.2, noEnumerateEnabled=False, full=True, multiple=False, timedProgress=30, silentFail=False, ddmin=False, swarmP=0.5, stopWhenBranches=None, noAlphaConvert=False, coverFile='coverage.out', readQuick=False, verboseActions=False, running=False, computeFeatureStats=False, compareFails=False, swarmProbs=None, output='failure.48845.test', genDepth=None, Pcrossover=0.2, quickAnalysis=False, keepLast=False, exploitCeiling=0.1, markovP=1.0, logging=None, stopWhenHitBranch=None, LOCProbs=False, stopWhenStatements=None, noCheck=False, maxTests=-1, noExceptionMatch=False, Pmutate=0.0, depth=200, trackMaxCoverage=None, throughput=False, noCover=True, timeout=60, stutter=None, uncaught=False, markov=None, showActions=False, startExploit=0.0)
UNCAUGHT EXCEPTION
ERROR: (<type 'exceptions.OSError'>, OSError(17, 'Directory exists in fake filesystem'), <traceback object at 0x1007c33b0>)
TRACEBACK:
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/sut.py", line 48329, in safely
    act[2]()
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/sut.py", line 66, in act0
    self.p_os[0] = newFakeOS()
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/sut.py", line 21, in newFakeOS
    fs.CreateDirectory('/Volumes')
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/pyfakefs/fake_filesystem.py", line 2057, in CreateDirectory
    raise OSError(errno.EEXIST, 'Directory exists in fake filesystem', directory_path)
Original test has 20 steps

SAVING TEST AS failure.48845.test
FINAL VERSION OF TEST, WITH LOGGED REPLAY:
str0 = ""                                                                # STEP 0
ACTION: str0 = "" 
str0 = None : <type 'NoneType'>
=> str0 = '' : <type 'str'>
==================================================
component4 = "g"                                                         # STEP 1
ACTION: component4 = "g" 
component4 = None : <type 'NoneType'>
=> component4 = 'g' : <type 'str'>
==================================================
component0 = "lambda"                                                    # STEP 2
ACTION: component0 = "lambda" 
component0 = None : <type 'NoneType'>
=> component0 = 'lambda' : <type 'str'>
==================================================
str0 += 'm'                                                              # STEP 3
ACTION: str0 += 'm' 
str0 = '' : <type 'str'>
=> str0 = 'm' : <type 'str'>
==================================================
component3 = "epsilon"                                                   # STEP 4
ACTION: component3 = "epsilon" 
component3 = None : <type 'NoneType'>
=> component3 = 'epsilon' : <type 'str'>
==================================================
component1 = "e"                                                         # STEP 5
ACTION: component1 = "e" 
component1 = None : <type 'NoneType'>
=> component1 = 'e' : <type 'str'>
==================================================
str0 += 'r'                                                              # STEP 6
ACTION: str0 += 'r' 
str0 = 'm' : <type 'str'>
=> str0 = 'mr' : <type 'str'>
==================================================
component2 = "lambda"                                                    # STEP 7
ACTION: component2 = "lambda" 
component2 = None : <type 'NoneType'>
=> component2 = 'lambda' : <type 'str'>
==================================================
str0 += 's'                                                              # STEP 8
ACTION: str0 += 's' 
str0 = 'mr' : <type 'str'>
=> str0 = 'mrs' : <type 'str'>
==================================================
str0 += 'y'                                                              # STEP 9
ACTION: str0 += 'y' 
str0 = 'mrs' : <type 'str'>
=> str0 = 'mrsy' : <type 'str'>
==================================================
str0 += 'o'                                                             # STEP 10
ACTION: str0 += 'o' 
str0 = 'mrsy' : <type 'str'>
=> str0 = 'mrsyo' : <type 'str'>
==================================================
str0 += 'j'                                                             # STEP 11
ACTION: str0 += 'j' 
str0 = 'mrsyo' : <type 'str'>
=> str0 = 'mrsyoj' : <type 'str'>
==================================================
str0 += 'b'                                                             # STEP 12
ACTION: str0 += 'b' 
str0 = 'mrsyoj' : <type 'str'>
=> str0 = 'mrsyojb' : <type 'str'>
==================================================
path2 = "/Volumes/ramdisk/test"                                         # STEP 13
ACTION: path2 = "/Volumes/ramdisk/test" 
path2 = None : <type 'NoneType'>
=> path2 = '/Volumes/ramdisk/test' : <type 'str'>
==================================================
path2 += "/" + component1                                               # STEP 14
ACTION: path2 += "/" + component1 
component1 = 'e' : <type 'str'>
path2 = '/Volumes/ramdisk/test' : <type 'str'>
=> path2 = '/Volumes/ramdisk/test/e' : <type 'str'>
==================================================
component1 = "b"                                                        # STEP 15
ACTION: component1 = "b" 
component1 = 'e' : <type 'str'>
=> component1 = 'b' : <type 'str'>
==================================================
bytes1 = 0                                                              # STEP 16
ACTION: bytes1 = 0 
bytes1 = None : <type 'NoneType'>
=> bytes1 = 0 : <type 'int'>
==================================================
str0 += 'm'                                                             # STEP 17
ACTION: str0 += 'm' 
str0 = 'mrsyojb' : <type 'str'>
=> str0 = 'mrsyojbm' : <type 'str'>
==================================================
bytes3 = 0                                                              # STEP 18
ACTION: bytes3 = 0 
bytes3 = None : <type 'NoneType'>
=> bytes3 = 0 : <type 'int'>
==================================================
os0 = newFakeOS()                                                       # STEP 19
ACTION: os0 = newFakeOS() 
os0 = None : <type 'NoneType'>
os_REF0 = None : <type 'NoneType'>
RAISED EXCEPTION: <type 'exceptions.OSError'> [Errno 17] Directory exists in fake filesystem: '/'
ERROR: (<type 'exceptions.OSError'>, OSError(17, 'Directory exists in fake filesystem'), <traceback object at 0x1007c3290>)
TRACEBACK:
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/sut.py", line 48329, in safely
    act[2]()
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/sut.py", line 66, in act0
    self.p_os[0] = newFakeOS()
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/sut.py", line 21, in newFakeOS
    fs.CreateDirectory('/Volumes')
  File "/Users/alex/tstl/examples/mutateProbe/pyfakefs/pyfakefs/fake_filesystem.py", line 2057, in CreateDirectory
    raise OSError(errno.EEXIST, 'Directory exists in fake filesystem', directory_path)
STOPPING TESTING DUE TO FAILED TEST
0.0378520488739 TOTAL RUNTIME
2 EXECUTED
220 TOTAL TEST OPERATIONS
0.00343918800354 TIME SPENT EXECUTING TEST OPERATIONS
0.0100502967834 TIME SPENT EVALUATING GUARDS AND CHOOSING ACTIONS
8.17775726318e-05 TIME SPENT CHECKING PROPERTIES
0.00352096557617 TOTAL TIME SPENT RUNNING SUT
0.00530886650085 TIME SPENT RESTARTING
0.0 TIME SPENT REDUCING TEST CASES
